editor/debugger/script_editor_debugger.cpp

```diff
@@ -345,6 +354,11 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, uint64_t p_thread
 			}
 			profiler->set_enabled(false, false);
 			visual_profiler->set_enabled(false);
+
+			if (restore_mouse_by_time > OS::get_singleton()->get_ticks_msec()) {
+				Input::get_singleton()->warp_mouse(restore_mouse_position);
```

Comment:
You are mixing different coordinate systems.
`get_global_mouse_position();` is in canvas layer coordinates.
`Input::get_singleton()->warp_mouse` is in screen coordinates.

I suggest to use the Control-functions instead, which are both in local coordinates:
``` c++
restore_mouse_position = get_local_mouse_position();
warp_mouse(restore_mouse_position);
```